src/build_scripts/buildlib.pio

```diff
@@ -187,13 +187,14 @@ def _set_pio_valid_values(case, valid_values):
     logger.warning("Updating valid_values for PIO_TYPENAME: {}".format(valid_values))
     env_run = case.get_env("run")
     env_run.set_valid_values("PIO_TYPENAME",valid_values)
-
+    
     for comp in case.get_values("COMP_CLASSES"):
         comp_pio_typename = "{}_PIO_TYPENAME".format(comp)
         current_value = case.get_value(comp_pio_typename)
         if current_value not in valid_values:
-            logger.warning("Resetting PIO_TYPENAME to netcdf for component {}".format(comp))
+            logger.warning("Resetting PIO_TYPENAME=netcdf and PIO_REARRANGER=1 for component {}".format(comp))
             env_run.set_value(comp_pio_typename,"netcdf")
+            env_run.set_value("PIO_REARRANGER", 1)
```

Comment:
Can you explain this change? It's definitely not intuitive; I'd like to understand this to get a slightly better understanding of how these pio settings are set. It seems like this block of code gets entered when trying to use pnetcdf but that doesn't exist on the machine, so the type is reset to netcdf. Your change to set the rearranger to 1 in this case suggests to me that you feel that the rearranger should be 1 whenever using type netcdf. But what I don't see is general logic that does that - i.e., if someone explicitly sets the type to netcdf, should there also be logic that sets the rearranger to 1? Or do you feel that's unnecessary?

Reply:
If the system determines that pnetcdf and netcdf4 are not available then the method is set to netcdf.  The rearranger=1 is more efficient in this case.  However if a user changes iotype to netcdf, I don't necessarily want to force them to also change the rearranger.  I suppose I could add a message in the code recommending changing pio_rearranger=1 - would you like that?

Reply:
Thanks @jedwards4b , this helps me. I feel like the best thing to do would be to add a few sentences of guidance wherever `PIO_REARRANGER` is defined in the xml configuration files. This guidance of generally using pio_rearranger=1 could be added to any other general guidance you can provide about when it's best to choose 1 vs. 2.

Reply:
I should add: I'm partly interested in this for the sake of CTSM's LILAC, where we are trying to set reasonable out-of-the-box PIO settings for users, regardless of their machine. So I'm wondering if a good rule of thumb is to use pio_rearranger=2 when using pnetcdf and pio_rearranger=1 when using vanilla netcdf, or if it's more complicated than that.